### PR TITLE
docs: fix references for the JOSS paper

### PR DESCRIPTION
**Pull request by sathvikbhagavan**

### CODE REVIEW

**Comment by sathvikbhagavan** [comment]
I could not find DOIs for three papers:

1. ModelingToolkit: A Composable Graph Transformation System For Equation-Based Modeling
2. On the stability of inverse problems by A. N. Tikhonov
3. Lectures on elementary mathematics, by Joseph Louis Lagrange; tr. by Thomas J. McCormack

Rest of them have been fixed.

File: joss/paper.bib [hunk 1]
```diff
@@ -96,19 +111,25 @@ @article{10.1145/321607.321609
...
...s {\}} {\backslash}right),{\$}{\$}corresponding to all integral values of the variable x = n. If these ordinates are the values of a known analytic function F(x), then the problem of interpolation between these ordinates has an obvious and precise meaning: we are required to compute intermediate values F(x) to the same accuracy to which the ordinates are known. Undoubtedly, the most convenient tool for the solution of this problem is the polynomial central interpolation method. It uses the polynomial of degree k --- 1, interpolating k successive ordinates, as an approximation to F(x) only within a unit interval in x, centrally located with respect to its k defining ordinates. Assuming k fixed, successive approximating arcs for F(x) are thus obtained which present discontinuities on passing from one arc to the next if k is odd, or discontinuities in their first derivatives if k is even (see section 2.121). Actually these discontinuities are irrelevant in our present case of an analytic function F(x). Indeed, if the interpolated values obtained are sufficiently accurate, these discontinuities will be apparent only if we force the computation beyond the intrinsic accuracy of the yn.",
```
Comment (ChrisRackauckas) [inline joss/paper.bib:123]: that's the worst abstract form ever 😅 who does a display block in abstract?!?!?!?!

**Comment by ChrisRackauckas** [comment]
ModelingToolkit: A Composable Graph Transformation System For Equation-Based Modeling is just on arxiv

File: joss/paper.bib (hunk 1 above)
Comment (sathvikbhagavan) [reply joss/paper.bib:123]: yeah, thats weird.... I took it from springer - https://link.springer.com/chapter/10.1007/978-1-4899-0433-1_1#citeas 😅

**Comment by sathvikbhagavan** [comment]
Ok, I am not familiar with the system, papers in arxiv don't assign DOI?

**Comment by ChrisRackauckas** [comment]
Arxiv has a DOI on the website, for example that one is https://doi.org/10.48550/arXiv.2103.05244

**Comment by sathvikbhagavan** [comment]
Ok, I have added the DOI for the MTK paper.

File: joss/paper.bib (hunk 1 above)
Comment (sathvikbhagavan) [reply joss/paper.bib]: I have removed it in [7dce110](https://github.com/SciML/DataInterpolations.jl/pull/265/commits/7dce1101f1edc8617cbba36b9c003a5eca6b28bf). The abstract is the first paragraph of the chapter, so not a real abstract.

**Comment by ChrisRackauckas** [comment]
The other two don't have DOIs.

**Pull request merged**